compiler/passes/src/dead_code_elimination/dead_code_eliminator.rs

```diff
@@ -14,23 +14,91 @@
 // You should have received a copy of the GNU General Public License
 // along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
 
-use leo_span::Symbol;
+use leo_ast::{AccessExpression, BinaryOperation, Expression, Node as _, Type, UnaryOperation};
+use leo_span::{Symbol, sym};
 
 use indexmap::IndexSet;
 
-#[derive(Default)]
-pub struct DeadCodeEliminator {
+use crate::TypeTable;
+
+#[derive(Debug)]
+pub struct DeadCodeEliminator<'a> {
```

Comment:
Is there a way to report some statistics on DCE so that a user has an idea of how much their program has changed?